backend/handlers/handlers_test.go

```diff
@@ -57,19 +98,179 @@ func TestListRoutes(t *testing.T) {
 		if len(methods) > 0 {
 			fmt.Print(" (" + strings.Join(methods, ",") + ")")
 		}
-		/* The following methods are only available in a newer version of mux: */
-		// queriesTemplates, err := route.GetQueriesTemplates()
-		// if err == nil {
-		// 	   fmt.Println("Queries templates:", strings.Join(queriesTemplates, ","))
-		// }
-		// queriesRegexps, err := route.GetQueriesRegexp()
-		// if err == nil {
-		// 	   fmt.Println("Queries regexps:", strings.Join(queriesRegexps, ","))
-		// }
 		fmt.Println()
 		return nil
 	})
 	if err != nil {
 		fmt.Println(err)
 	}
 }
+
+func requestTester(request *http.Request, t *testing.T) *httptest.ResponseRecorder {
+	connectionData := handlers.NewConnectionData(8082, "")
+	backendInstance, err := backend.NewBackend(arguments.NewArguments(
+		test.TstTempDir("test-tempdir-"), false, false, false, false, false),
+		webdevEnvironment{},
+	)
+	if err != nil {
+		t.Errorf("An error occurred. %v", err)
+	}
+
+	handlers := handlers.NewHandlers(backendInstance, connectionData)
+
+	rr := httptest.NewRecorder()
+	handlers.Router.ServeHTTP(rr, request)
+	return rr
+}
+
+func checkResponse(t *testing.T, response *httptest.ResponseRecorder, expectedResponse string) {
+	if status := response.Code; status != http.StatusOK {
+		t.Errorf("Status code differs. Expected %d. Got %d", http.StatusOK, status)
+	}
+	if response.Body.String() != expectedResponse {
+		t.Errorf("Get Testing response body differs. Expected %q. Got %q", expectedResponse, response.Body.String())
+	}
+}
+
+func TestConfigDefaultHandler(t *testing.T) {
+	req, err := http.NewRequest("GET", "/api/config/default", nil)
+	if err != nil {
+		t.Errorf("An error occurred. %v", err)
+	}
+	rr := requestTester(req, t)
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("Status code differs. Expected %d. Got %d", http.StatusOK, status)
+	}
+
+	req, err = http.NewRequest("POST", "/api/config", bytes.NewBuffer(rr.Body.Bytes()))
+	if err != nil {
+		t.Errorf("An error occurred. %v", err)
+	}
+	rr = requestTester(req, t)
+	if status := rr.Code; status != http.StatusOK {
+		t.Errorf("Status code differs. Expected %d. Got %d", http.StatusOK, status)
+	}
+}
+
+func TestNotifyHandler(t *testing.T) {
+	req, err := http.NewRequest("POST", "/api/notify-user", bytes.NewBuffer([]byte("{\"text\":\"test\"}")))
+	if err != nil {
+		t.Errorf("An error occurred. %v", err)
+	}
+
+	rr := requestTester(req, t)
+	checkResponse(t, rr, "null\n")
+}
+
+func TestOpenHandler(t *testing.T) {
+	req, err := http.NewRequest("POST", "/api/open", bytes.NewBuffer([]byte("\"malformed\"")))
+	if err != nil {
+		t.Errorf("An error occurred. %v", err)
+	}
+
+	rr := requestTester(req, t)
+	checkResponse(t, rr, "{\"error\":\"Blocked /open with url: malformed\"}\n")
```

Comment:
this is not unit testing the handlers, but unit testing the backend, as the backend produces error.

What you want is to check that backend.SystemOpen() was actually called with the right argument, and mock a response. For this you want to pass your own implementation the backend interface and mock it.

Tools like `mockery` help to autogenerate those, and testify's mocking helps to easily mock.

Example to generate:
https://github.com/digitalbitbox/bitbox-wallet-app/blob/b5b67c42f42eb13d4f33925d2773b77ecbf4ac18/backend/coins/btc/headers/headers.go#L52
Example to use:
https://github.com/digitalbitbox/bitbox-wallet-app/blob/master/backend/coins/btc/transactions/transactions_test.go#L131 

Same applies for other tests here.

It's not high prio to fix this, and it's also an effort, but in the spirit of unit testing, here you only want to test that the handlers do the correct input parsing, the correct backend call, and produce the correct output. The actual logic should be unit tested in backend_test.go